### ISSUE TITLE
[coc.nvim] Connection to server got closed. Server will not be restart.

### ISSUE THREAD

**Issue by dreampasssser** [neoclide/coc-eslint#152]
When I opened a `ts` file, an error was reported. This is eslint output:

```
[Info  - 20:14:03] ESLint server is starting.
node:internal/modules/cjs/loader:1228
  throw err;
  ^

Error: Cannot find module 'F:\Users\11024\AppData\Local\coc\extensions\node_modules\coc-eslint\lib\server.js'
    at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
    at Function._load (node:internal/modules/cjs/loader:1055:27)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5)
    at node:internal/main/run_main_module:36:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v22.14.0
[Info  - 20:14:04] Connection to server got closed. Server will restart.
[Info  - 20:14:04] ESLint server stopped.
[Info  - 20:14:04] ESLint server is starting.
[Error - 20:14:04] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 20:14:04] ESLint server stopped.
[Error - 20:14:04] ESLint client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097
```

Then, I execute `:CocCommand eslint.restart`, but `[coc.nvim] Restarting client failed` was reported.

Here is `Cocinfo`:

```
## versions

vim version: VIM - Vi IMproved 9.1 9011251
node version: v22.14.0
coc.nvim version: 0.0.82-22130a1 2025-04-23 00:26:27 +0800
coc.nvim directory: C:\Users\11024\vimfiles\plugged\coc.nvim
term: undefined
platform: win32

## Log of coc.nvim

2025-04-24T20:16:14.334 INFO (pid:22332) [extension:coc-git] - Looking for git in: git
2025-04-24T20:16:14.598 INFO (pid:22332) [plugin] - coc.nvim initialized with node: v22.14.0 after 1375
2025-04-24T20:16:14.604 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:16:15.154 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:16:18.298 INFO (pid:22332) [language-client-index] - Language server "eslint" started with 22364
2025-04-24T20:16:18.354 INFO (pid:22332) [services] - service tsserver started
2025-04-24T20:16:18.413 ERROR (pid:22332) [server] - unhandledRejection  Promise {
  <rejected> _ResponseError: Pending response rejected since connection got disposed
      at Object.dispose (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:20634:25)
      at Object.dispose (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:74802:31)
      at LanguageClient.handleConnectionClosed (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:75723:30)
      at LanguageClient.handleConnectionClosed (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:76223:15)
      at closeHandler (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:75710:16)
      at CallbackList.invoke (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:18793:35)
      at _Emitter.fire (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:18852:34)
      at closeHandler (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:19844:24)
      at CallbackList.invoke (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:18793:35)
      at _Emitter.fire (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:18852:34) {
    code: -32097,
    data: undefined
  }
} _ResponseError: Pending response rejected since connection got disposed
    at Object.dispose (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:20634:25)
    at Object.dispose (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:74802:31)
    at LanguageClient.handleConnectionClosed (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:75723:30)
    at LanguageClient.handleConnectionClosed (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:76223:15)
    at closeHandler (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:75710:16)
    at CallbackList.invoke (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:18793:35)
    at _Emitter.fire (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:18852:34)
    at closeHandler (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:19844:24)
    at CallbackList.invoke (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:18793:35)
    at _Emitter.fire (C:\Users\11024\vimfiles\plugged\coc.nvim\build\index.js:18852:34) {
  code: -32097,
  data: undefined
}
2025-04-24T20:16:18.420 INFO (pid:22332) [language-client-index] - Language server "eslint" started with 19488
2025-04-24T20:16:18.533 INFO (pid:22332) [language-client-index] - Language server "eslint" started with 5272
2025-04-24T20:16:18.614 INFO (pid:22332) [language-client-index] - Language server "eslint" started with 20840
2025-04-24T20:16:19.742 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:19:58.430 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:19:59.112 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:19:59.716 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:20:00.521 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:20:02.352 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:20:03.054 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:20:15.351 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:20:16.825 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:20:18.932 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:20:19.897 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:20:24.515 INFO (pid:22332) [attach] - receive notification: showInfo []
2025-04-24T20:20:25.014 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:20:54.728 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:21:00.291 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:21:01.098 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:21:01.799 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:21:04.292 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:21:06.342 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:21:08.392 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:21:09.059 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:21:29.335 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:21:53.166 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:22:31.267 INFO (pid:22332) [attach] - receive notification: runCommand [ 'eslint.restart' ]
2025-04-24T20:22:31.614 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:29:14.583 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:29:15.960 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:29:16.713 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:29:20.640 INFO (pid:22332) [attach] - receive notification: highlight []
2025-04-24T20:29:24.583 INFO (pid:22332) [attach] - receive notification: showInfo []

```

**Comment by chemzqm** [comment]
>  'F:\Users\11024\AppData\Local\coc\extensions\node_modules\coc-eslint\lib\server.js'

That file should exist, reinstall by `:CocInstall coc-eslint` or try manually remove the folder `F:\Users\11024\AppData\Local\coc\extensions\node_modules\coc-eslint` first.

**Comment by dreampasssser** [comment]
@chemzqm I tried, it didn't work.

**Comment by dreampasssser** [comment]
@chemzqm Emm... It was installed in `C:`, but why searched in `F:`.  I'm confused.

**Comment by chemzqm** [comment]
Could be in your runtimepath, checkout `:CocList extensions` in your vim.

**Comment by dreampasssser** [comment]
I executed the commond, here is the result:

![Image](https://github.com/user-attachments/assets/7ffbf8f4-a6b7-47f5-9047-c4ff73bcef8f)

I cannot detect anything.

My project is in `F:` and pnpm store is also in `F:`. I'm not sure if this matters or not. But other extentions work well.

**Comment by chemzqm** [comment]
Does the folder `F:\Users\11024\AppData\Local\coc\extensions\node_modules\coc-eslint` exists?

> My project is in F: and pnpm store is also in F:. I'm not sure if this matters or not. But other extentions work well.

Not related, my suggestion is try to remove `F:\Users\11024\AppData\Local\coc\extensions` completely.

**Comment by dreampasssser** [comment]
Doesn't exist. There is not even a Users folder in `F:`. My system is installed in `C:`.

**Comment by chemzqm** [comment]
Seems something wrong with your `URI.parse`, made a fix might helps.

**Comment by dreampasssser** [comment]
OK👍, it's fixed, thank you.
